File: js-budget-api/src/models/transaction.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Transaction extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
      Transaction.belongsTo(models.Credential, {
        onDelete: "CASCADE",
        foreignKey: {
          field: "credentialId",
          allowNull: false,
          name: "credentialId",
        },
      });
    }
  }
  Transaction.init(
    {
      id: {
        type: DataTypes.UUID,
        primaryKey: true,
        defaultValue: DataTypes.UUIDV4,
      },
      credentialId: DataTypes.UUID,
      category: DataTypes.STRING(50),
      description: DataTypes.STRING(200),
      value: DataTypes.DECIMAL,
      timestamp: DataTypes.DATE(3),
    },
    {
      sequelize,
      modelName: "Transaction",
    },
  );
  return Transaction;
};

File: go-budget-api/internal/models/transaction.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Transaction struct {
	Id uuid.UUID `gorm:"type:uuid;primary_key"`
	Category string `gorm:"type:varchar(50);NOT NULL"`
	Description string `gorm:"type:varchar(200);NOT NULL"`
	Value float32 `gorm:"type:decimal;NOT NULL"`
	Timestamp time.Time `gorm:"type:timestamp(3);NOT NULL"`
	Credential Credential `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
	CredentialId uuid.UUID `gorm:"NOT NULL"`
}

func (t *Transaction) BeforeCreate(tx *gorm.DB) (err error) {
	t.Id = uuid.New()
	return
   }
